src/Neo/SmartContract/ApplicationEngine.Runtime.cs

```diff
@@ -428,8 +428,18 @@ private static bool CheckItemType(StackItem item, ContractParameterType type)
                     return aType == StackItemType.Integer;
                 case ContractParameterType.ByteArray:
                 case ContractParameterType.String:
-                    return aType is StackItemType.Any or StackItemType.ByteString or StackItemType.Buffer &&
-                        Utility.StrictUTF8.GetString(item.GetSpan()) is not null; // Prevent any non-UTF8 string
+                    {
+                        if (aType is StackItemType.Any or StackItemType.ByteString or StackItemType.Buffer)
+                        {
```

Comment:
Now this check is being performed for both cases when `type` is either  `ContractParameterType.ByteArray` or `ContractParameterType.String`. But the initial idea was to perform this check only for `ContractParameterType.String` and let the `ContractParameterType.ByteArray` be as is.

Reply:
My bad :)